tests/bwrite.rs

```diff
@@ -17,128 +17,554 @@
+        "local",
+        "bwrite",
+        "--input",
+        &batch_input_file_path,
+    ])
+    .output()?;
 
-    // more verification would be nice
-    assert_eq!(
-        true,
-        predicate::str::is_match("\"Dimensions\":")?.eval(String::from_utf8(output)?.as_str())
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
+    );
+
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name2,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
     );
 
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_batch_write_put() -> Result<(), Box<dyn std::error::Error>> {
+    let mut tm = util::setup().await?;
+    let table_name = tm.create_temporary_table("pk", None).await?;
+
+    let mut c = tm.command()?;
+    c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "bwrite",
+        "--put",
+        r#"{"pk": "11",
+        "null-field": null,
+        "list-field": [1, 2, 3, "str"],
+        "map-field": {"l0": <<1, 2>>, "l1": <<"str1", "str2">>, "l2": true},
+        "binary-field": b"\x00",
+        "binary-set-field": <<b"\x01", b"\x02">>}"#,
+    ])
+    .output()?;
+
     let mut c = tm.command()?;
     let get_cmd = c.args(&[
         "--region",
         "local",
         "--table",
         &table_name,
         "get",
-        "ichi",
+        "11",
         "-o",
         "raw",
     ]);
-
     let output = get_cmd.output()?.stdout;
     let data: Value = serde_json::from_str(&String::from_utf8(output)?)?;
+    assert_eq!(data["pk"]["S"], "11");
+    assert_eq!(data["null-field"]["NULL"], true);
+    assert_eq!(data["list-field"]["L"][0]["N"], "1");
+    assert_eq!(data["list-field"]["L"][1]["N"], "2");
+    assert_eq!(data["list-field"]["L"][2]["N"], "3");
+    assert_eq!(data["list-field"]["L"][3]["S"], "str");
+    assert_eq!(data["map-field"]["M"]["l0"]["NS"][0], "1");
+    assert_eq!(data["map-field"]["M"]["l0"]["NS"][1], "2");
+    assert_eq!(data["map-field"]["M"]["l1"]["SS"][0], "str1");
+    assert_eq!(data["map-field"]["M"]["l1"]["SS"][1], "str2");
+    assert_eq!(data["map-field"]["M"]["l2"]["BOOL"], true);
+    assert_eq!(
+        data["binary-field"]["B"],
+        general_purpose::STANDARD.encode(b"\x00")
+    );
+    assert_eq!(
+        data["binary-set-field"]["BS"][0],
+        general_purpose::STANDARD.encode(b"\x01")
+    );
+    assert_eq!(
+        data["binary-set-field"]["BS"][1],
+        general_purpose::STANDARD.encode(b"\x02")
+    );
```

Comment:
Can we utilize `assert_eq_json`?

Reply:
Same as https://github.com/awslabs/dynein/pull/190#discussion_r1535927712